### PR TITLE
Update on API Documentation

### PR DESCRIPTION
**Pull request by tfrauenstein**
Some editorial improvements on API documentation — differ between reference definition and user manual documentation.

### CODE REVIEW

File: general-guidelines/GeneralGuidelines.md [hunk 1]
```diff
@@ -2,18 +2,31 @@
...
-## {{ book.must }} API First: Define APIs using OpenAPI
+## {{ book.must }} Provide API Reference Definition using OpenAPI
 
 API First is one of our architecture principles, as per our Architecture Rules of Play.
 Define, document and review your APIs before delivering them (also see API review procedure).
-Whether you choose YAML or JSON as a format of your specification file is up to you.
+Whether you choose YAML or JSON as a format of your OpenAPI API specification file is up to you; 
+however, YAML is generally preferred due to its improved readability. 
+We also call the OpenAPI API specification the "API Reference Definition"; 
+it provides all information needed by an experienced API user developer to use this API.
```
Comment (fmueller) [inline general-guidelines/GeneralGuidelines.md]: minor editorial improvement: API user developer -> client developer (you use same term in user manual documentation rule)

Comment (tfrauenstein) [inline general-guidelines/GeneralGuidelines.md]: done

**Comment by fmueller** [comment]
👍

**Comment by tkrop** [comment]
:+1:

**Pull request merged**